### PR TITLE
[feature] Enable custom re-usable sessions for API.

### PR DESCRIPTION
**Pull request by johnchildren**
Adds a session attribute to the API wrapper class which can be supplied when objects of that class are initialized. This is useful when setting custom headers for the session or setting custom retry policies.

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/api_wrappers.py [hunk 1]
```diff
@@ -90,6 +90,7 @@ def __init__(
         use_websocket: bool = True,
         provider: Optional[str] = None,
         support_mfa: bool = True,
+        session: Optional[Session] = None,
```
Comment (cqc-melf) [inline pytket/extensions/quantinuum/backends/api_wrappers.py:93]: Could you please add this to the documentation?

Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/api_wrappers.py:93]: (And maybe to the changelog in `docs/changelog.rst` )

Comment (johnchildren) [reply pytket/extensions/quantinuum/backends/api_wrappers.py:93]: done!

Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/api_wrappers.py:93]: Thank you!

**Review by cqc-melf — APPROVED** [review]
Looks good, thank you.

**Pull request merged**